test/Microsoft.AspNet.Razor.Test/TagHelpers/TagHelperBlockRewriterTest.cs

```diff
@@ -0,0 +1,909 @@
+                        new MarkupBlock(
+                            new MarkupTagHelperBlock(
+                                "input",
+                                selfClosing: true,
+                                attributes: new List<KeyValuePair<string, SyntaxTreeNode>>()
+                                {
+                                    new KeyValuePair<string, SyntaxTreeNode>("class", expression),
+                                    new KeyValuePair<string, SyntaxTreeNode>("unbound-required", null),
+                                })),
+                        noErrors
+                    },
+                    {
+                        $"<p class='{expressionString}' bound-string></p>",
+                        new MarkupBlock(
+                            new MarkupTagHelperBlock(
+                                "p",
+                                selfClosing: false,
+                                attributes: new List<KeyValuePair<string, SyntaxTreeNode>>()
+                                {
+                                    new KeyValuePair<string, SyntaxTreeNode>("class", expression),
+                                    new KeyValuePair<string, SyntaxTreeNode>("bound-string", null),
+                                })),
+                        new[]
+                        {
+                            new RazorError(
+                                string.Format(errorFormat, "bound-string", "p", stringType),
+                                absoluteIndex: 29,
+                                lineIndex: 0,
+                                columnIndex: 29,
+                                length: 12)
+                        }
+                    },
+                    {
+                        "<input bound-required-string class='btn' />",
+                        new MarkupBlock(
+                            new MarkupTagHelperBlock(
+                                "input",
+                                selfClosing: true,
+                                attributes: new List<KeyValuePair<string, SyntaxTreeNode>>()
+                                {
+                                    new KeyValuePair<string, SyntaxTreeNode>("bound-required-string", null),
+                                    new KeyValuePair<string, SyntaxTreeNode>("class", factory.Markup("btn")),
+                                })),
+                        new[]
+                        {
+                            new RazorError(
+                                string.Format(errorFormat, "bound-required-string", "input", stringType),
+                                absoluteIndex: 7,
+                                lineIndex: 0,
+                                columnIndex: 7,
+                                length: 21)
+                        }
+                    },
+                    {
+                        $"<input bound-required-string class='{expressionString}' />",
```

Comment:
Why is it necessary to test `class='{expressionString}'` in combination with almost every other attribute type?